### DISCUSSION TITLE
在项目中使用原版仙剑奇侠传1资源文件的问题

### DISCUSSION THREAD

**Discussion by rexzhang** [py-pal/py-pal#4] [Ideas]
# 如果要在项目直接使用`原版仙剑奇侠传1`的资源包文件
- 需要实现专门的资源文件的访问接口库,而不只是单纯的理解文件格式
- 内部的资源文件,大概率大部分不是 `.png` `.jpg` 等通用的格式,需要使用这些资源的代码需要解决格式转换问题, `on fly` 就有性能问题
- 因为打算对部分资源做高清化处理和替代,这就需要另外一套资源访问机制;两种机制混合使用的技术成本太高

# 建议的方案
- 选择 文件夹/资源包(可以兼容补充文件夹) 中的任意一种为项目的资源文件的存储方式
- 理解`原版仙剑奇侠传1`的资源包文件(简称:原版资源包)格式后,用代码实现一次性导出+格式转换的功能
- 自定义的资源体系中尽量使用对执行(绘制/播放)代码更友好的格式
- 程序分发时不包含原版资源包文件(版权问题)
- 玩家在第一次运行时,提示提供原版资源包所在路径,导入后再开始游戏
  - 实现完整的资源替代后,不需要导入也可以游戏,类似OpenTTD
  - 只要设计的项目资源存储格式支持叠加和 patch, 就可以很灵活的支持这个效果

**Comment by alanshi** [comment]
**关于原版资源的问题：**

我们在开发过程中肯定会对原版的资源做部分高清处理， 不管是资源访问机制还是打包机制都会改变，再兼容之前的资源文件成本确实很高，我赞同提供资源格式导出转换的工具给用户，对原版资源进行处理后再进行游戏，只是这个地方是否涉及到版权问题，需要考虑。

**Comment by rexzhang** [comment]
如果完全不转换格式,直接调用,应该没有法律问题,很多开源项目都是这么做的.这个方式实际上是玩家通过购买游戏购买了使用权.
提供转换工具导入就不晓得,这个需要法律支援.
OpenTTD是在后期实现了自己的资源库标准和高清资源,而且社区各种不同版本的资源特别多

**Comment by rexzhang** [comment]
OpenTTD对原版资源的使用方式就是保留格式;还有一个CorsixTH(主题医院);keeperfx(地下城守护者).你可以都下载下来看看.其中OpenTTD我觉得是最成功的,已经是一个非常强大的社区了

**Comment by rexzhang** [comment]
刚刚研究了下原版资源包大概有30MB,如果全部将所有非一次性资源装载到内存应该不会超过300MB.启动时全部转换并装载到内存也不是不可以.然后定义自己的格式,逐步填充没有版权问题的资源也是一种方法
